packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -54,36 +60,33 @@ export class Communicator {
       window.addEventListener('message', listener);
       this.listeners.set(listener, { reject });
     });
-  }
-
-  private listeners = new Map<(_: MessageEvent) => void, { reject: (_: Error) => void }>();
+  };
 
   /**
-   * Rejects all requests and clears the listeners
+   * Closes the popup, rejects all requests and clears the listeners
    */
-  disconnect() {
+  private disconnect = () => {
+    closePopup(this.popup);
```

Comment:
Should the SDK close the popup or just set `this.popup = null`?

Reply:
good discussion.
sdk should only set it to null.